cpp/CMakeLists.txt

```diff
@@ -247,9 +247,14 @@ if(${CLANG_TIDY_FOUND})
                     ${ARROW_LINT_QUIET})
 endif()
 
-if(ARROW_ONLY_LINT)
```

Comment:
This broke IWYU and clang-tidy because `compile-commands.json` wasn't generated. The impact should be minimal since we changed the defaults to minimal dependencies.